File: stack.cpp
#include<iostream>
using namespace std;
int main(){
    int stack[4];
    int a,top=0;
    for (int i=0;i<4;i++){
        cout<<"enter data-";
        cin>>stack[top++];
        
    }
    for(int i=0;i<4;i++){
        cout<<stack[--top]<<"\n";
    }
    return 0;
}

File: stack.java
import java.util.Scanner;

public class stack {
    public static void main(String args[]){
        int stack[]=new int[4];
        int top=-1;
        Scanner Scan= new Scanner(System.in);
        for (int i=0;i<4;i++){
            stack[++top]=Scan.nextInt();
        }
        for(int i=0;i<4;i++){
            System.out.println(stack[top--]);
        }
        //return 0;
    }
    
}
